src/TableGateway/AbstractTableGateway.php

```diff
@@ -189,32 +191,35 @@ public function select($where = null)
 
         if ($where instanceof \Closure) {
             $where($select);
-        } elseif ($where !== null) {
+        } elseif ($where !== NULL) {
             $select->where($where);
         }
 
-        return $this->selectWith($select);
+        return $this->selectWith($select, $params);
     }
 
     /**
      * @param Select $select
-     * @return null|ResultSetInterface
-     * @throws \RuntimeException
+     * @param array  $params - statement params
+     *
+     * @return null|\Zend\Db\ResultSet\ResultSetInterface
      */
-    public function selectWith(Select $select)
+    public function selectWith(Select $select, array $params = [])
```

Comment:
Not true. You can add optional arguments to implementations; it then becomes a matter of the user understanding whether or not they are using the interface signature, or the implementation signature. Considering the majority of TableGatewayInterface implementations start from this abstract class, it's fine — but we should likely add a todo item to the interface to add these new arguments in a new major version.